### PR TITLE
Ignore generated files in jacoco coverage

### LINKED ISSUE #87: exclude from Jacoco coverage
Jacoco exclusion works by having` @Generated` annotation in bytecode, but the default `javax.annotation.processing.Generated` in only on source level.
https://github.com/jacoco/jacoco/blob/046ba52e9fd1288cf121fd1013c9019d71946954/org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/AnnotationGeneratedFilter.java

Can I add a custom annotation inside record-builder?

### PR DESCRIPTION
**Pull request by madisparn**
Fixes #87

### CODE REVIEW

File: record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java [hunk 1]
```diff
@@ -571,35 +619,7 @@ public String p2() {
             });
             witherClassNameBuilder.add(">");
         }
-        var witherClassName = witherClassNameBuilder.build().toString();
-        var codeBuilder = CodeBlock.builder()
-                .add("return new $L", witherClassName)
```
Comment (madisparn) [inline record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java:576]: Jacoco does not check anonymous class annotations, so changing to `new @RecordBuilderGenerated With() {}` did not exclude the inner class from coverage

**Comment by Randgalt** [comment]
Thanks for the submission - I'll review soon

File: record-builder-core/src/main/java/io/soabase/recordbuilder/core/RecordBuilder.java [hunk 2]
```diff
@@ -94,6 +94,11 @@
          */
         String fromMethodName() default "from";
 
+        /**
+         * The name to use for the from-to-wither inner class
+         */
+        String fromInnerClassName() default "WitherImpl";
```
Comment (Randgalt) [inline record-builder-core/src/main/java/io/soabase/recordbuilder/core/RecordBuilder.java:100]: Can you explain this better? I'm -1 on this if its only purpose is to make Jacoco happy.

Comment (madisparn) [reply record-builder-core/src/main/java/io/soabase/recordbuilder/core/RecordBuilder.java:100]: It is not about making Jacoco happy, it is about not including generated code in test coverage.

Comment (Randgalt) [reply record-builder-core/src/main/java/io/soabase/recordbuilder/core/RecordBuilder.java:100]: Can you explain in detail please?

Comment (madisparn) [reply record-builder-core/src/main/java/io/soabase/recordbuilder/core/RecordBuilder.java:100]: image
I would like the record-builder generated classes to be excluded from coverage.

Comment (Randgalt) [reply record-builder-core/src/main/java/io/soabase/recordbuilder/core/RecordBuilder.java:100]: OK - but can't this be done by configuring Jacoco? Changing the code generation just so code coverage produces better reports seems backwards to me.

Comment (madisparn) [reply record-builder-core/src/main/java/io/soabase/recordbuilder/core/RecordBuilder.java:100]: To me it seems better if library users have to configure their build as little as possible. Excluding generated code from test coverage seems like it should be supported by annotation processor. Lombok does it.

**Comment by Randgalt** [comment]
I see you closed this. I'll try to spend time on this to see if there's a good solution that meets both of our goals.

**Comment by Randgalt** [comment]
I didn’t either. Can you reopen please?

**Comment by madisparn** [comment]
Did not see a way to open this PR, created new one https://github.com/Randgalt/record-builder/pull/96

**Pull request closed without merging**